ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ProxyHandler.java

```diff
@@ -134,7 +134,18 @@ private static void responseHandler(final HttpClientResponse clientResponse,
     Pipe.pipe(clientResponse, serverResponse, () -> {
       final MultiMap trailers = clientResponse.trailers();
       serverResponse.trailers().setAll(trailers);
-    });
+    }, () -> checkFirstWrite(clientResponse, serverResponse));
+  }
+
+  // Vert.x requires that content-length is set if not chunked but Jetty sometimes sends
+  // response with no content-length and not chunked, so we workaround this by adding a
+  // tranfer encoding response header with chunked in this case
+  private static void checkFirstWrite(final HttpClientResponse clientResponse,
+      final HttpServerResponse serverResponse) {
+    if (clientResponse.getHeader("content-length") == null
+        && clientResponse.getHeader("transfer-encoding") == null) {
+      serverResponse.putHeader("transfer-encoding", "chunked");
```

Comment:
Is the response actually formatted as if it were a chunked response (length of current chunk followed by carriage return, newline, content, etc.)? It's not, right? Then how come this workaround doesn't result in issues decoding the server response from the client that made the request?

Reply:
Chunked transfer encoding must always be acceptable on a client as per spec: (see 14.3.9): https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html